flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/common/config/BigQueryConnectOptions.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.common.config;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.util.function.SerializableSupplier;
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.flink.bigquery.services.BigQueryServices;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Objects;
+
+/** BigQuery client connection configuration. */
+@AutoValue
+@PublicEvolving
+public abstract class BigQueryConnectOptions implements Serializable {
+
+    public abstract String getProjectId();
+
+    public abstract String getDataset();
+
+    public abstract String getTable();
+
+    @Nullable
+    public abstract CredentialsOptions getCredentialsOptions();
+
+    @Nullable
+    public abstract SerializableSupplier<BigQueryServices> getTestingBigQueryServices();
+
+    @Override
+    public final String toString() {
+        return String.format(
+                "[project : %s, dataset : %s, table : %s]",
+                getProjectId(), getDataset(), getTable());
+    }
+
+    @Override
+    public final int hashCode() {
+        int hash = 5;
+        hash = 61 * hash + Objects.hashCode(getProjectId());
+        hash = 61 * hash + Objects.hashCode(getDataset());
+        hash = 61 * hash + Objects.hashCode(getTable());
+        hash = 61 * hash + Objects.hashCode(getCredentialsOptions());
+        return hash;
+    }
+
+    @Override
+    public final boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        final BigQueryConnectOptions other = (BigQueryConnectOptions) obj;
+        if (!Objects.equals(this.getProjectId(), other.getProjectId())) {
```

Comment:
We could merge these lines together following the same format in spark-bigquery-connector.
```
return Objects.equals(this.getProjectId(), other.getProjectId())
  && Objects.equals(this.getDataset(), other.getDataset())
  && Objects.equals(this.getTable(), other.getTable())
  && Objects.equals(this.getCredentialsOptions(), other.getCredentialsOptions());
```

Reply:
Same comment as previous, I'll change it.